homeassistant/components/ebusd/__init__.py

```diff
@@ -0,0 +1,118 @@
+"""
+Support for Ebusd daemon for communication with eBUS heating systems.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/ebus/
+"""
+
+from datetime import timedelta
+import logging
+import socket
+
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_NAME, CONF_HOST, CONF_PORT, CONF_MONITORED_CONDITIONS)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.discovery import load_platform
+from homeassistant.util import Throttle
+
+from .const import (DOMAIN, SENSOR_TYPES)
+
+REQUIREMENTS = ['ebusdpy==0.0.16']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'ebusd'
+DEFAULT_PORT = 8888
+CONF_CIRCUIT = 'circuit'
+CACHE_TTL = 900
+SERVICE_EBUSD_WRITE = 'ebusd_write'
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=15)
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_CIRCUIT): cv.string,
+        vol.Required(CONF_HOST): cv.string,
+        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_MONITORED_CONDITIONS, default=[]): vol.All(
+            cv.ensure_list, [vol.In(SENSOR_TYPES['700'])])
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, config):
+    """Set up the eBusd component."""
+    conf = config[DOMAIN]
```

Comment:
Please use [] style for keys that are required or have a default.

Reply:
DOMAIN is a constant value so will never be null

Reply:
Sorry, this ended up on the wrong line. It was supposed to be marking the next two.

Reply:
CONF_NAME has a default on config schema above and CONF_CIRCUIT is required so it should be already managed and both should be never get without a value. Am I correct?

Reply:
You are correct. Our convention in that case is to use the [] syntax and not the .get syntax.

```
name = conf.get(CONF_NAME)
circuit = conf.get(CONF_CIRCUIT)
```

would change to:

```
name = conf[CONF_NAME]
circuit = conf[CONF_CIRCUIT]
```